src/test/java/org/springframework/boot/autoconfigure/data/jdbc/JdbcRepositoriesMyBatisAutoConfigurationTests.java

```diff
@@ -47,11 +51,29 @@ public void tearDown() {
 	@Test
 	public void myBatisDefaultsWork() {
 
-		this.context = prepareApplicationContext(JdbcRepositoriesAutoConfigurationTests.TestConfiguration.class);
+		this.context = prepareApplicationContext( //
+				JdbcRepositoriesAutoConfigurationTests.TestConfiguration.class, //
+				MyBatisTestConfiguration.class);
 
-		assertThat(this.context.getBean(DataAccessStrategy.class)).isInstanceOf(MyBatisDataAccessStrategy.class);
+		DataAccessStrategy dataAccessStrategy = this.context.getBean(DataAccessStrategy.class);
+		List strategies = getField(dataAccessStrategy, "strategies");
+		assertThat(strategies).hasSize(2);
 
```

Comment:
I would favor peeking at the collection and asserting that `MyBatisDataAccessStrategy` is in there.